File: Reactivities-API/Reactivities.Domain/Activities/Dtos/ActivityDto.cs
namespace Reactivities.Domain.Activities.Dtos
{
    public class ActivityDto
    {
        public required long Id { get; set; }
        public required string Title { get; set; }
        public required DateTime Date { get; set; }
        public string? Description { get; set; }
        public required string Category { get; set; }
        public required string City { get; set; }
        public required string Venue { get; set; }
        public required bool IsCancelled { get; set; }
        public required AttendeeDto Host { get; set; }
        public List<AttendeeDto> Attendees { get; set; } = new();
    }
}

File: WebClient/src/app/models/dtos/activityDto.ts
import Attendee from '../../../models/attendee'

export default interface ActivityDto {
  id: number
  title: string
  date: Date
  description?: string
  category: string
  city: string
  venue: string
  isCancelled: boolean
  host: Attendee
  attendees: Attendee[]
}
